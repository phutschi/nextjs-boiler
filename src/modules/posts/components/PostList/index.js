// @flow

import * as React from 'react';
import classNames from 'classnames';

import theme from './main.scss';

type Props = { }

const PostList = ( {} : Props ) => {
    const mainClassnames = classNames( theme.postList );

    return (
        <div className={mainClassnames}>
            {children}
        </div>
    );
};

PostList.defaultProps = { };

export default PostList;
