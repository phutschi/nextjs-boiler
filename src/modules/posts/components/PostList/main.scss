.postList {
  margin: 0;
}

.postListItemContainer {
  margin: 0;
}
