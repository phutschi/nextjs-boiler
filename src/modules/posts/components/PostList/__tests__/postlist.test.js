/* global expect */

import * as React from 'react';

import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import PostList from '../';
// import { Loading } from '../../../../components/components';

const posts = [ {
    id    : 1,
    title : 'First Blog post',
}, {
    id    : 2,
    title : 'Second Blog Post',
} ];

describe( '<PostList />', () => {
    it( 'Snapshot', () => {
        const wrapper = renderer.create( <PostList posts={posts} /> ).toJSON();

        expect( wrapper ).toMatchSnapshot();
    } );

    it( 'Renders Loading while loading posts', () => {
        const wrapper = shallow( <PostList loading posts={posts} /> );

        expect( wrapper.find( 'Loading' ).length ).toEqual( 1 );
    } );

    it( 'Renders all posts', () => {
        const wrapper = shallow( <PostList posts={posts} /> );

        expect( wrapper.find( '.postListItemContainer' ).length ).toEqual( 1 );
        expect( wrapper.find( '.postListItemContainer' ).children().length ).toEqual( posts.length );
    } );
} );
