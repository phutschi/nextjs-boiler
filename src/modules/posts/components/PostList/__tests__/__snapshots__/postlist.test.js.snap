// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PostList /> Snapshot 1`] = `
<div
  className="postList"
>
  <div
    className="postListItemContainer"
  >
    <div>
      First Blog post
    </div>
    <div>
      Second Blog Post
    </div>
  </div>
</div>
`;
